src/main/java/edu/umich/its/cpm/EmailFormatter.java

```diff
@@ -74,6 +140,53 @@ public String getBody() {
         return prunedHeaders;
     }
 
+    public ArrayList<String> mboxFormatHeaders() {
+        ArrayList<String> headers = prunedHeadersWithoutContentType();
+        String startingHeader = mboxStartingHeader(headers);
+        headers.add(0, startingHeader);
+        return headers;
+    }
+
+    public String mboxStartingHeader(ArrayList<String> headers) {
+        String from = null;
+        String date = null;
+        StringBuilder startingHeader = new StringBuilder();
+        for (String header : headers) {
+            // From: Jane Doe <jdoe@umich.edu>
+            if (header.startsWith(FROM_WITH_COLON)) {
+                from = header.split(FROM_WITH_COLON)[1].trim();
+                if (from.indexOf('<') != -1) {
+                    from = from.substring(from.indexOf('<') + 1, from.indexOf('>'));
+                }
+            }
+//             Date: Wed, 24 Aug 2016 10:04:47 -0400
+            if (header.startsWith(DATE_WITH_COLON)) {
+                String dateTemp = header.split(DATE_WITH_COLON)[1].trim();
+                try {
+                    DateFormat localTimeFormat = new SimpleDateFormat("EEE, d MMM yyyy HH:mm:ss Z");
+                    Date localDateAndTime = localTimeFormat.parse(dateTemp);
+                    long epochTime = localDateAndTime.getTime();
+                    DateFormat ascTimePattern = new SimpleDateFormat("EEE MMM dd HH:mm:ss yyyy");
+                    ascTimePattern.setTimeZone(TimeZone.getTimeZone("UTC"));
+                    // Wed Aug 24 14:04:47 2016
+                    date = ascTimePattern.format(new Date(epochTime));
+                } catch (java.text.ParseException e) {
+                    log.error("Error occurred while paring the Date " + e.getMessage());
+                }
+
+            }
+
+            if (date == null) {
```

Comment:
`date == null` case occur when `localTimeFormat.parse(dateTemp);`fail so in case of this situation i felt i don't need to stop the mbox formatting process. I can use the current date in asctime format in the Starting header having it won't hurt the mbox migration as this date that i am formatting is going to be tossed away by the mbox reader anyway. So in short its one way of error handling.